server/index.js

```diff
@@ -55,6 +55,89 @@ app.get('/api/products/:productId', (req, res, next) => {
 
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    return res.status(200).json([]);
+  }
+  const getAllCartItems = `
+  select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+  db.query(getAllCartItems, [req.session.cartId])
+    .then(response => res.status(200).json(response.rows));
+
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({ error: 'productId must be a positive integer' });
+  }
+  const checkForPriceSQL = `
+  select "price"
+  from "products"
+  where "productId" = $1;
+  `;
+
+  db.query(checkForPriceSQL, [productId])
+    .then(response => {
+      if (!response.rows[0]) {
+        next(new ClientError(`a product with ID: ${productId} does not exist`, 404));
+      }
```

Comment:
You should `throw new ClientError(message, statusCode)` instead of calling `next()` here. We want to shortcut to the `.catch()` and calling `next()` won't stop this promise chain.